### PR TITLE
boards: arm: Modify BMD-345-EVAL support

### PR DESCRIPTION
**Pull request by ghost**
Modify support for u-blox BMD-345-EVAL which uses the nRF52840
and a Skyworks RFX2411 FEM.

These edits follow the naming cnoventions that is used
with the other u-blox EVKs recently added, or in progress

This board is similar to the nRF52840dk_nrf52840 with the
addition of a FEM. Four Arduino GPIO pins have been
reassigned to the PA_LNA control pins.

u-blox would prefer to use this naming convention to match
other BMD-3xx-EVAL and EVK-NINA-Bx boards recently submitted.

Tested with blinky, button, and Bluetooth peripheral_hr

Signed-off-by: Bob Recny <bob.recny@u-blox.com>

### CODE REVIEW

**Comment by brec-u-blox** [comment]
@Navin-Sankar: The Compliance check failed on a CODEOWNERS error. Since you made the original PR with the original naming, is this something you can modify?

**Comment by Navin-Sankar** [comment]
@brec-u-blox change the board name in CODEOWNERS file also.

File: boards/arm/ubx_bmd345eval_nrf52840/Kconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# BMD-345-EVAL board configuration
+
+# Copyright (c) 2021 u-blox AG
+# Copyright (c) 2021 Linumiz
+# SPDX-License-Identifier: Apache-2.0
+
+# No special configuration is required.
```
Comment (Navin-Sankar) [inline boards/arm/ubx_bmd345eval_nrf52840/Kconfig:7]: nrf52840 chip itself have on board DC to DC regulator. Its need to be enabled to handle low power modes.

Comment (brec-u-blox) [reply boards/arm/ubx_bmd345eval_nrf52840/Kconfig:7]: The nRF52840 has a DC-DC converter, but it requires external inductors in order to use it. The BMD-345 module does not have these inductors and no means to add them due to the space required on the module for the RFX2411 FEM. Enabling the DC-DC converter will cause an application to fail. Only the default LDO within the nRF52840 may be used.

Comment (Navin-Sankar) [reply boards/arm/ubx_bmd345eval_nrf52840/Kconfig:7]: if I understand correctly, there is no LC circuit between the DCCH, VDD and DCC, DEC4. That's why we don't need to enable the DC/DC regulator.

Comment (brec-u-blox) [reply boards/arm/ubx_bmd345eval_nrf52840/Kconfig:7]: Correct

**Review by Navin-Sankar — APPROVED** [review]

File: CODEOWNERS [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# CODEOWNERS for autoreview assigning in github
+# CODEOWNERS for autoreview assigning in githubbmd
```
Comment (ioannisg) [inline CODEOWNERS:1]: is this intentional @brec-u-blox ?

Comment (brec-u-blox) [reply CODEOWNERS:1]: ugh - This must have crept in when searching in nano and thought I hit ^w first. I'll fix.

**Review by ioannisg — APPROVED** [review]

File: boards/arm/ubx_bmd345eval_nrf52840/board.c [hunk 3]
```diff
@@ -11,7 +12,7 @@
 #define MODE_PIN	36	/* P1.04 */
 #define A_SEL_PIN	34	/* P1.02 */
```
Comment (anangl) [inline boards/arm/ubx_bmd345eval_nrf52840/board.c:13]: These values are used as parameters in calls to `gpio_pin_configure()` so they should be 4 and 2, respectively. If it works in the current form, it only works by chance. This assertion: https://github.com/zephyrproject-rtos/zephyr/blob/0500d75c3a9ad228867529a650147534dc027e1c/include/drivers/gpio.h#L689-L690 should actually fail for both those calls.

Comment (brec-u-blox) [reply boards/arm/ubx_bmd345eval_nrf52840/board.c:13]: I seem to be missing a CMakeLists.txt file, so board.c never got pulled into my tests, and the default state is low.
Working through this now. This is not returning a valid device:
```        mode_asel_port_dev = device_get_binding(DT_LABEL(DT_NODELABEL(gpio1)));```

Comment (brec-u-blox) [reply boards/arm/ubx_bmd345eval_nrf52840/board.c:13]: Converted this to draft mode until this is sorted.

**Comment by brec-u-blox** [comment]
Fixes implemented. Ready for review.

**Review by anangl — APPROVED** [review]

**Comment by brec-u-blox** [comment]
Just checking status. Can this be merged?

**Pull request merged**